### PR TITLE
Generic get and update service, updated for the author controller

### PR DESCRIPTION
**Pull request by unnamedsunshine**
updated the generic service for  get and update 
updated the author controller accordingly

### CODE REVIEW

File: backend/src/controller/author.ts [hunk 1]
```diff
@@ -44,6 +44,24 @@ const getAllAuthors = async (req: Request, res: Response) => {
...
+    ).get(authorId);
+    res.json(result);
+  } catch (error: any) {
+    if (error instanceof NotFoundError) {
+      res.status(404).send(error.message);
+    } else {
+      res.status(500).send(error.message);
+    }
+  }
+};
```
Comment (Maudfer) [inline backend/src/controller/author.ts:60]: I like it, the code is very clean, easy to read. Nice!

File: backend/src/service/generic.ts [hunk 2]
```diff
@@ -82,8 +84,53 @@ export function genericService<T>(
     }
   };
 
+  /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (Maudfer) [inline backend/src/service/generic.ts]: If you are using these to make the code easier to read, I think there are some vscode extensions that help you visually separate the functions when looking at them, :)
It's probably best to remove these before merging. I don't hgave anything against it personally, but it's not very standard hahaha

File: backend/src/service/generic.ts [hunk 3]
```diff
@@ -82,8 +84,53 @@ export function genericService<T>(
...
+      query
+    );
+
+    if (result.rows.length == 0) {
+      throw new NotFoundError("Not found.");
+    } else {
+      const got = new ModelClass(result.rows[0]);
+      return got;
+    }
+  };
```
Comment (Maudfer) [inline backend/src/service/generic.ts:60]: Nice and clean! Great code!
I'd just cleanup the comments before merging, if all is working :)

**Review by Maudfer — APPROVED** [review]

**Comment by Maudfer** [comment]
Approved! Added some nitpicks, but great code other than that!

File: backend/src/service/generic.ts (hunk 2 above)
Comment (ideheer) [reply backend/src/service/generic.ts]: Yeah. I agree with Mauricio.

**Review by ideheer — APPROVED** [review]

**Pull request merged**